lib/producer.js

```diff
@@ -295,16 +294,18 @@ Producer.prototype.disconnect = function(timeout, cb) {
  *
  * Initialize transactions, this is only performed once per transactional producer.
  *
- * @param {number} timeout - Number of milliseconds to try to initialize before giving up.
- *                           defaults to 5 seconds.
+ * @param {number} timeout - Number of milliseconds to try to initialize before giving up, defaults to 5 seconds.
+ * @param {function} cb - Callback to return when operation is completed
  * @return {Producer} - returns itself.
  */
- Producer.prototype.initTransactions = function(timeout) {
-  if (timeout === undefined || timeout === null) {
+Producer.prototype.initTransactions = function(timeout, cb) {
+  if (typeof timeout === 'function') {
     timeout = 5000;
+    cb = timeout;
```

Comment:
This shall be done before assign before `timeout=5000`

Reply:
Good catch